sisl/viz/plots/bands.py

```diff
@@ -433,41 +434,52 @@ def _read_from_wfsx(self, root_fdf, wfsx_file, extra_vars=()):
         self.geometry = fdf.read_geometry(output=True)
 
         # Get the wfsx file
-        wfsx_sile = self.get_sile(wfsx_file or "wfsx_file")
+        wfsx_sile = self.get_sile(wfsx_file or "wfsx_file", parent=self.geometry)
```

Comment:
It is not clear to me in this PR whether PDOS or `eigenstate` will be called with parent being the geometry? If so you'll get wrong values...

This goes for both fatbands and PDOS.

Reply:
I didn't know `norm2` used the overlap matrix :scream: Indeed I was not getting the hamiltonian for fatbands. For the PDOS I already set the hamiltonian as the parent. Thanks for noticing!

Reply:
Everytime you are doing something with the eigenvectors you generally need the overlap matrix due to the eigenvectors not being orthogonal. It is only for the wavefunction where the overlap is not needed (it is implicitly calculated during the construction of the real space grid).

Reply:
Fixed, now fatbands from `WFSX` files require the `Hamiltonian` to be set as parent. And if it doesn't find one, then it fails.